userappwithbroker/src/main/java/com/microsoft/aad/adal/example/userappwithbroker/MainActivity.java

```diff
@@ -292,23 +296,28 @@ public void onError(Exception exc) {
      */
     private void saveUserIdFromAuthenticationResult(final AuthenticationResult authResult, final String authority) {
         mSharedPreference = getSharedPreferences(SHARED_PREFERENCE_STORE_USER_UNIQUEID, MODE_PRIVATE);
+        if (null != authResult) {
+            final SharedPreferences.Editor prefEditor = mSharedPreference.edit();
+            if (null != authResult.getAuthority()) {
```

Comment:
so if authority comes back as null, shouldn't you raise hell here saying this should not happen? Just add a Toast

Reply:
Addressed.